pandas/tests/io/formats/test_format.py

```diff
@@ -1269,18 +1269,60 @@ def test_to_string_specified_header(self):
             df.to_string(header=['X'])
 
     def test_to_string_no_index(self):
-        df = DataFrame({'x': [1, 2, 3], 'y': [4, 5, 6]})
 
-        df_s = df.to_string(index=False)
-        expected = "x  y\n1  4\n2  5\n3  6"
+        dfs = [
 
-        assert df_s == expected
+            # ints
+            DataFrame({'x': [1, 2, 3], 'y': [4, 5, 6]}),
+            DataFrame({'x': [11, 22, 33], 'y': [4, 5, 6]}),
+            DataFrame({'x': [11, 22, -33], 'y': [4, 5, 6]}),
+            DataFrame({'x': [11, 22, -33], 'y': [4, 5, -6]}),
+            DataFrame({'x': [11, 22, -33], 'y': [44, 55, -66]}),
+
+            # floats
+            DataFrame({'x': [0.1, 0.2, -0.3], 'y': [4, 5, 6]}),
+            DataFrame({'x': [0.1, 0.2, -0.3], 'y': [0.4, 0.5, 0.6]}),
+            DataFrame({'x': [0.1, 0.2, -0.3], 'y': [0.4, 0.5, -0.6]}),
+        ]
```

Comment:
In cases like this is preferred to use pytest parametrization.

Reply:
Will fix if/when general direction of this PR is accepted.